coderd/database/dbmem/dbmem.go

```diff
@@ -6952,7 +6953,9 @@ func (q *FakeQuerier) GetWorkspacesEligibleForTransition(ctx context.Context, no
 		if user.Status == database.UserStatusActive &&
 			job.JobStatus != database.ProvisionerJobStatusFailed &&
 			build.Transition == database.WorkspaceTransitionStop &&
-			workspace.AutostartSchedule.Valid {
+			workspace.AutostartSchedule.Valid &&
+			(workspace.NextStartAt.Time.IsZero() ||
```

Comment:
Suggestion: Can we can add a note here that if we check zero so that we can update it and that it does't mean the workspace is actually eligible for transition?